compiler/rustc_errors/src/lib.rs

```diff
@@ -521,12 +521,24 @@ impl Handler {
     }
 
     /// Construct a builder at the `Warning` level at the given `span` and with the `msg`.
+    /// Cancel the builder if warnings cannot be emitted
     pub fn struct_span_warn(&self, span: impl Into<MultiSpan>, msg: &str) -> DiagnosticBuilder<'_> {
         let mut result = self.struct_warn(msg);
         result.set_span(span);
         result
     }
 
+    /// Construct a builder at the `Warning` level at the given `span` and with the `msg`.
```

Comment:
I found this comment confusing-- I expected it to comment on the "force" part of the method name here.

Reply:
What does it mean to *force* a warning?

Reply:
I'll rework this. I think all these functions might need better comments.